### DISCUSSION TITLE
Use SQLModel

### DISCUSSION THREAD

**Discussion by davidbrochart** [fastapi-users/fastapi-users#698] [Q&A]
[SQLModel](https://github.com/tiangolo/sqlmodel) was just released. Do you see any interest in using it with FastAPI Users?

**Comment by frankie567** [comment]
Definitely. I even think this could supersede SQLAlchemy 1.4 support.

I'm currently in the process of splitting the database backends in several packages to avoid conflicts in dependencies. Once this is done, we'll be able to implement new backends.

**Reply by glendigity** [reply]
Awesome. Thank you! Excited about using them together!

**Reply by SSardorf** [reply]
Do you have an approximate ETA on this?

**Reply by frankie567** [reply]
Not really, I'm struggling a bit to get the implementation right... Problem is I have to support both standard Pydantic models and SQLModel models, which causes loads of issues...

**Reply by frankie567** [reply]
I'm also blocked by https://github.com/tiangolo/sqlmodel/issues/25

**Reply by hacen-ai** [reply]
Hello @frankie567, I think it was fixed by using UUID.hex

**Reply by frankie567** [reply]
Yes, but we need the fix to be merged in the main branch before moving on 🙂

**Reply by davidbrochart** [reply]
https://github.com/tiangolo/sqlmodel/pull/26 was merged and released in v0.0.7! :smile:

**Reply by xavipolo** [reply]
I can't get it to work :worried:
Does anyone know if there are plans to make a tutorial or add some documentation? :pray:

Thanks.

**Reply by frankie567** [reply]
@xavipolo I don't plan currently to officially support it. But basically the model could look like this:

```py
from fastapi_users_db_sqlmodel import SQLModelBaseUserDB


class User(SQLModelBaseUserDB, table=True):
    __tablename__ = "user"

    class Config:
        orm_mode = True
```

And the `UserManager` would be defined like this:

```py
class UserManager(UUIDIDMixin, BaseUserManager[User, UUID4]):
    ...
```

**Reply by xavipolo** [reply]
Thanks, It's running ok now 👍

**Comment by unidesigner** [comment]
It seems as there is now the https://github.com/fastapi-users/fastapi-users-db-sqlmodel database adapter for this, but there is no documentation yet. Would this solve this issue?
